tests/components/eafm/test_sensor.py

```diff
@@ -0,0 +1,431 @@
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        },
+    )
+
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+    assert state.attributes["unit_of_measurement"] == "m"
+
+    await poll(
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        }
+    )
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "unavailable"
+
+    await poll(
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        }
+    )
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+
+
+async def test_unload_entry(hass, mock_get_station):
+    """Test being able to unload an entry."""
+    entry, _ = await async_setup_test_fixture(
+        hass,
+        mock_get_station,
+        {
+            "label": "My station",
+            "measures": [
+                {
+                    "@id": "really-long-unique-id",
+                    "label": "York Viking Recorder - level-stage-i-15_min----",
+                    "qualifier": "Stage",
+                    "parameterName": "Water Level",
+                    "latestReading": {"value": 5},
+                    "stationReference": "L1234",
+                    "unit": "http://qudt.org/1.1/vocab/unit#Meter",
+                    "unitName": "m",
+                }
+            ],
+        },
+    )
+
+    # And there should be an entity
+    state = hass.states.get("sensor.my_station_water_level_stage")
+    assert state.state == "5"
+
+    assert await entry.async_unload(hass)
+
+    # And the entity should be gone
+    assert not hass.states.get("sensor.my_station_water_level_stage")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)